Module/hints.py

```diff
@@ -32,50 +43,93 @@ def generateHints(locationItems, hintsType, seedName, outZip):
                         hintedWorlds.append(world)
 
 
-            if locationType.HUNDREDAW in hintedWorlds:
+            if pages_need_hints:
                 for world in worldChecks:
                     if not world in hintedWorlds and any(item.ItemType == itemType.TORN_PAGE for item in worldChecks[world]):
                         hintedWorlds.append(world)
 
 
-            if locationType.Atlantica in hintedWorlds:
+            if mag_thun_need_hints:
                 for world in worldChecks:
-                    if not world in hintedWorlds and any(item.ItemType == itemType.THUNDER or item.ItemType == itemType.MAGNET for item in worldChecks[world]):
+                    if not world in hintedWorlds and any(item.ItemType == itemType.THUNDER for item in worldChecks[world]):
                         hintedWorlds.append(world)
 
-
-            for world in hintedWorlds:
-                random.shuffle(reportsList)
-                reportNumber = reportsList.pop()
-                hintsText["Reports"][reportNumber] = {
-                    "World": world,
-                    "Count": len(worldChecks[world]),
-                    "Location": ""
-                }
-
-            hintedHints = []
-
-            for reportNumber in hintsText["Reports"].keys():
+            if mag_thun_need_hints:
                 for world in worldChecks:
-                    if not world in hintedWorlds and not world in hintedHints and any(item.Name.replace("Secret Ansem's Report ","") == str(reportNumber) for item in worldChecks[world] ):
-                        hintedHints.append(world)
-                        hintsText["Reports"][reportNumber]["Location"] = world
+                    if not world in hintedWorlds and any(item.ItemType == itemType.MAGNET for item in worldChecks[world]):
+                        hintedWorlds.append(world)
 
-            for world in hintedHints:
-                random.shuffle(reportsList)
-                reportNumber = reportsList.pop()
+            # hintedWorlds is now all the required hinted worlds. We'll see if we can also hint the reports that hint proofs
+            if len(hintedWorlds) > 13:
+                hintedWorlds = hintedWorlds[0:13]
+
+            random.shuffle(reportOrdering)
+            proof_report_order = None
+            for ordering in reportOrdering:
+                remaining_report_slots = 13-len(hintedWorlds)
+                worlds_to_add = []
+                for index,reportNumber in enumerate(ordering):
+                    invalid = False
+                    for world in worldChecks:
+                        if any(item.Name.replace("Secret Ansem's Report ","") == str(reportNumber) for item in worldChecks[world] ):
+                            #don't allow a report for a proof to be locked by that proof
+                            if (proof_of_peace_index is index and isReportOnMushroom[reportNumber]) or \
```

Comment:
Turns out I got lucky in the test seeds I generated. If `reportNumber == 13`, the reportNumber is out of the bounds of the list, since reports are 1-indexed, but the list is 0-indexed. This also would likely cause an issue where the incorrect report would get blacklisted from the hint logic. Feel free to correct this and resubmit the PR!

Reply:
lol, I genned 60000 seeds with a tester file and didn't crash, but I guess that was just luck. I'll fix and double check the indexing